### PR TITLE
Fix/bug 21294

### PR DESCRIPTION
**Pull request by srkunze**

### CODE REVIEW

**Comment by srkunze** [comment]
Should I change if `ret['retcode'] != 0` to `if ret['retcode'] != 0` as well? Or do you prefer the `>` notation as it is done in function `create`.

I preferred `!=` since the value space of`retcode` actually has no natural ordering.

File: salt/modules/cmdmod.py [hunk 1]
```diff
@@ -1397,6 +1397,28 @@ def exec_code(lang, code, cwd=None):
     return stdout
```
Comment (jfindlay) [inline salt/modules/cmdmod.py]: I think we should have `exec_code` call `exec_code_all`, 
```python
    ret = exec_code_all(lang, code, cwd)
    return ret['stdout']
```

Comment (srkunze) [inline salt/modules/cmdmod.py]: Done.

**Comment by jfindlay** [comment]
Nice, thanks @srkunze.

**Comment by thatch45** [comment]
Yes != is the right way to eval retcodes since they can technically be negative.
I will come back with a deeper eval here when I can

**Comment by jfindlay** [comment]
@thatch45, also note that @srkunze [has indicated](https://github.com/saltstack/salt/issues/21294#issuecomment-81816766) that he'll close this and resubmit against 2015.2.

**Pull request closed without merging**